pkg/registry/core/pod/storage/storage.go

```diff
@@ -245,6 +247,13 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types
 		for k, v := range annotations {
 			pod.Annotations[k] = v
 		}
+		// Copy all labels from the Binding over to the Pod object, but do not
+		// overwrite any existing labels that have been previously set, to avoid
+		// changing any existing behaviour for pods that may be defined with a
+		// template by users and created by higher-level controllers.
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodTopologyLabelsAdmission) {
+			copyLabelsWithoutOverwriting(pod, labels)
```

Comment:
name suggests there will be no overrides while the design saying we will override: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4742-node-topology-downward-api#design-details

Am I missing some change in design or the name is misleading here?

/hold

Reply:
I had adapted this to *not* overwrite values to ensure a deterministic result through admission, i.e. to permit other admission plugins to override our built-in plugin's decision on the zone/region value.

You are right though, that this shouldn't extend to the storage layer, as this would prevent admission plugins overwriting existing labels (which _is_ the intended design/expectation).

I've just pushed up an extra commit + integration test to address this. PTAL!

Reply:
/unhold

Reply:
Copying comment for viz: I see the change to overwrite, but I am not sure it's the correct thing to do. We can't go back and fix annotations, but why are we sure this won't break someone?